modin/core/storage_formats/cudf/query_compiler.py

```diff
@@ -54,6 +59,17 @@ def iloc_mut(partition, row_internal_indices, col_internal_indices, item):
                     partition.iloc[i, j] = it
             return partition
 
+        if not is_scalar(item):
+            broadcasted_item, _ = broadcast_item(
+                self,
+                row_numeric_index,
+                col_numeric_index,
+                item,
+                need_columns_reindex=need_columns_reindex,
+            )
+        else:
+            broadcasted_item, _ = item, pandas.Series([np.array(item).dtype])
```

Comment:
same question as in base QC - what's the point?

Reply:
fixed